app/controllers/CyaRequestAdviceController.scala

```diff
@@ -56,27 +59,33 @@ class CyaRequestAdviceController @Inject() (
             ).flatten
           )
           Ok(view(list, recordId))
-        case Left(errors) => logErrorsAndContinue(errors, recordId)
+        case Left(errors) => logErrorsAndContinue(errors, recordId, request)
       }
   }
 
-  def logErrorsAndContinue(errors: data.NonEmptyChain[ValidationError], recordId: String): Result = {
+  def logErrorsAndContinue(
+    errors: data.NonEmptyChain[ValidationError],
+    recordId: String,
+    request: DataRequest[AnyContent]
+  ): Result = {
     val errorMessages = errors.toChain.toList.map(_.message).mkString(", ")
 
     val continueUrl = RedirectUrl(routes.AdviceStartController.onPageLoad(recordId).url)
 
     logger.warn(s"Unable to create Request Advice.  Missing pages: $errorMessages")
+    dataCleansingService.deleteMongoData(request.userAnswers.id)
```

Comment:
We don't seem to be awaiting or handling the Future[Boolean] returned from dataCleansingService -> Could this cause issues or not?

Reply:
If this call fails, we would take the user to an error page instead of journey recovery but the session data would still be there...I think it's ok for this to be a fire and forget but haven't thought about all the pros/cons

Reply:
yeah just fire and forget same like audit service!